src/App.js

```diff
@@ -33,6 +37,43 @@ function App() {
   function selectValue(event) {
     setSelect(event);
   }
+
+  function addToShoppingCard(card) {
+    const objectExists = localStorage.find((object) => object.id === card.id);
+
+    if (objectExists) {
+      setLocalStorage(
+        localStorage.map((obj) =>
+          obj.id === card.id
+            ? { ...objectExists, amount: objectExists.amount + 1 }
+            : obj
+        ) /* wenn es existiert-dann ändere amount mit .map, wenn nein-gib obj unverändert zurück  */
+      );
+    } else {
+      setLocalStorage([...localStorage, { ...card, amount: 1 }]);
+    }
+    navigate("/warenkorb");
+  }
+
+  function decreaseAmount(obj) {
+    console.log(obj);
+    console.log(obj.amount);
+
+    if (obj.amount === 0) {
+      deleteCard(obj);
+    } else {
+      setLocalStorage(
+        localStorage.map((item) =>
+          obj.id === item.id ? { ...obj, amount: obj.amount - 1 } : item
+        )
+      );
+    } /* warum nicht functioniert mit delete hier???? */
```

Comment:
Explanation to your comment:
If you delete the item here, then you're ignoring the amount value. You'd be deleting the item if it were 5, 10, or 20.
But you only want to delete it if it's down to zero, which is why it's correct to put the delete function in the if-case of amount === 0.

Reply:
hey Ernst, sorry i forgot to delete the comment after fixing the bug. i fixed the mistake and it works now. It did not work because in this ternary operator instead of ": item" at the and i put ":obj". They mean the same but  obj comes from outside as props and is not a result of localStorage.map (but item is)  But thank you!